src/main/kotlin/com/devfloor/untitled/comment/application/CommentService.kt

```diff
@@ -1,15 +1,26 @@
 package com.devfloor.untitled.comment.application
 
-import com.devfloor.untitled.article.domain.Article
+import com.devfloor.untitled.article.application.ArticleService
 import com.devfloor.untitled.comment.domain.Comment
 import com.devfloor.untitled.comment.domain.CommentRepository
+import com.devfloor.untitled.common.exception.EntityNotFoundException
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 
 @Service
 class CommentService(
-    private val commentRepository: CommentRepository,
+    private val repository: CommentRepository,
+    private val articleService: ArticleService,
 ) {
-    fun showAllByArticle(article: Article): List<Comment> {
-        return commentRepository.findAllByArticle(article)
+    @Transactional(readOnly = true)
+    fun showByCommentId(commentId: Long): Comment = repository.findByIdOrNull(commentId)
+        ?: throw EntityNotFoundException("존재하지 않는 댓글입니다.")
+
+    @Transactional
+    fun modifyByArticleId(articleId: Long, commentId: Long, request: CommentModifyRequest) {
+        val article = articleService.showById(articleId)
+        val comment = showByCommentId(commentId)
+        comment.modify(Comment(article, comment.author, comment.anonymous, request.content))
     }
```

Comment:
```suggestion
    @Transactional
    fun modifyByArticleId(articleId: Long, commentId: Long, request: CommentModifyRequest) {
        val article = articleService.showById(articleId)
        showByCommentId(commentId).apply {
            modify(...)
        }
    }
```

근데 이거 리뷰달다가 갑자기 든 생각이 comment에 각각의 PK가 존재한다면 굳이 articleId가 필요가 없을 것 같은데?
url도 `comments/{id}`로 가져가도 될 것 같은데? 이건 좀 이야기를 해봐야할듯?

Reply:
수정사항은 반영해서 수정하겠습니다.
ㅇㅇ 어차피 코멘트 조회해와서 내용 수정하는데 코멘트에 article은 변할 일이 없어서 ..
코멘트 한개를 수정 삭제하는 함수에선 필요없을듯